### PR TITLE
SC-3303: Align BCBox with V0.6 file upload changes

### PR DESCRIPTION
**Pull request by wilwong89**
# Description
Update BCBox to use new API for COMS.
Change createObject and updateObject methods to use PUT instead of POST
Upload as octet-stream

## Types of changes

Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

## Further comments
Made an assumption that objects passed to objService will be of `File` type.

### CODE REVIEW

File: frontend/src/utils/utils.ts [hunk 1]
```diff
@@ -62,3 +62,20 @@ export function partition<T>(
     [[], []] as [Array<T>, Array<T>]
   );
 }
+
+/**
+ * @function setDispositionHeader
+ * Constructs a value for http request header 'Content-Disposition'
```
Comment (jujaga) [inline frontend/src/utils/utils.ts:68]: Constructs a valid RFC 6266 'Content-Disposition' header and optionally handles RFC 8187 UTF-8 encoding when necessary

File: frontend/src/services/objectService.ts [hunk 2]
```diff
@@ -295,8 +301,7 @@ export default {
       );
     }
 
-    const fd = new FormData();
-    fd.append('file', object);
-    return comsAxios(axiosOptions).post(`${PATH}/${objectId}`, fd, config);
+    const fd = await object.arrayBuffer();
```
Comment (jujaga) [inline frontend/src/services/objectService.ts:304]: I'm not completely certain what the await does, but does this end up forcing the browser to consume a large file payload (multiple gigs) and shove into memory if we end up awaiting? Just slightly concerned about potential adverse effects to browser memory usage on significantly large arrayBuffered files.

File: frontend/src/services/objectService.ts [hunk 3]
```diff
@@ -26,7 +27,10 @@ export default {
     axiosOptions?: AxiosRequestConfig
   ) {
     const config = {
-      headers: { 'Content-Type': 'multipart/form-data' },
+      headers: {
+        'Content-Type': 'application/octet-stream',
+        'Content-Disposition': setDispositionHeader(object.name)
+      },
```
Comment (jujaga) [inline frontend/src/services/objectService.ts:33]: Two things:
1. Please alphabetize the headers - Disposition should come before Type
2. We default to mimetype `application/octet-stream`, but we can't say every file is of this type. Please take a look at the `file-type` npm package or equivalent to have it inspect the incoming arrayBuffer stream and determine what kind of mimetype it is.

File: frontend/src/services/objectService.ts [hunk 4]
```diff
@@ -274,7 +277,10 @@ export default {
...
```
Comment (jujaga) [inline frontend/src/services/objectService.ts:283]: Two things:
1. Please alphabetize the headers - Disposition should come before Type
2. We default to mimetype `application/octet-stream`, but we can't say every file is of this type. Please take a look at the `file-type` npm package or equivalent to have it inspect the incoming arrayBuffer stream and determine what kind of mimetype it is.

**Review by jujaga — COMMENTED** [review]
Definitely glad this pathway actually works with axios without too much relative headache. That being said, we need to make sure we're actually doing some proper handling of the mimetype detection and not just calling everything an octet-stream.

File: frontend/src/utils/utils.ts (hunk 1 above)
Comment (wilwong89) [reply frontend/src/utils/utils.ts]: Added

File: frontend/src/services/objectService.ts (hunk 4 above)
Comment (wilwong89) [reply frontend/src/services/objectService.ts]: Content type removed, will resolve in another ticket.

File: frontend/src/services/objectService.ts (hunk 2 above)
Comment (wilwong89) [reply frontend/src/services/objectService.ts:305]: After testing on some images, it doesn't seem like there's a synchronous lock on the browser while it uploads. But that isn't a full test on browser memory and possible adverse effects.

**Review by jujaga — APPROVED** [review]

**Review by norrisng-bc — APPROVED** [review]

File: frontend/src/services/objectService.ts [hunk 5]
```diff
@@ -28,16 +28,21 @@ export default {
...
         tagset: {}
       },
     };
 
+    // Detect MIME type from File object and assign to header
+    // Fall back to application/octet-stream as above
+    if (object?.type) {
+      config.headers['Content-Type'] = object.type;
+    }
+
```
Comment (jujaga) [inline frontend/src/services/objectService.ts:45]: We can collapse the mimetype logic down into an in-stream one-liner
```ts
headers: {
  'Content-Disposition': setDispositionHeader(object.name),
  'Content-Type': object?.type ? object.type : 'application/octet-stream'
},
```

File: frontend/src/services/objectService.ts [hunk 6]
```diff
@@ -279,14 +284,20 @@ export default {
...
```
Comment (jujaga) [inline frontend/src/services/objectService.ts:300]: We can collapse the mimetype logic down into an in-stream one-liner
```ts
headers: {
  'Content-Disposition': setDispositionHeader(object.name),
  'Content-Type': object?.type ? object.type : 'application/octet-stream'
},
```

**Pull request merged**